### PR TITLE
Jetpack: Reorganize My Site Menu

### PR DESCRIPTION
**Pull request by salimbraksa**
## Requirement 

Re-organize My Site menu items into 3 sections:

- **Content**, the bits of information published on a site.
- **Traffic**, the flow and study of visitors viewing a site.
- **Maintenance**, the ongoing things that help keep a site running smoothly.

Source pcdRpT-1Vl-p2

## Implementation Details

1. The `My Site > Menu` screen data is stored in an array called `tableSections` of type `NSArray<BlogDetailsSection>`. And each `BlogDetailsSection` object has a list of rows of type `NSArray<BlogDetailsRow>`.
2. The method setting the `tableSections` array is `configureTableViewData`. So this menu reorganization is done by refactoring the `configureTableViewData` method.
3. These changes are only applied to Jetpack. See [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/20476/files#r1157857588).
4. Each section ( Content, Traffic, and Maintenance ) is a method in `BlogDetailsViewController`.
    - -(BlogDetailsSection *)contentSectionViewModel
    - -(BlogDetailsSection *)trafficSectionViewModel
    - -(BlogDetailsSection *)maintenanceSectionViewModel
5. The rows implementation was embedded in the existing `sectionViewModel` methods ( see `- (BlogDetailsSection *)jetpackSectionViewModel` method for example ). So to avoid code duplication, I extracted those rows into properties of `BlogDetailsViewController` and re-used them in the new `sectionViewModel` methods.

## Screenshots

| Before | After |
| ------- | ----- |
|  |  |

## Test Instructions

I recommend testing this PR by comparing the new menu with the old one and ensuring the same rows are visible. To do that, either get 2 devices and build & run this PR in one of the devices or install the **Jetpack Alpha** build alongside the one from AppStore.

Please repeat the following steps using:
- a wordpress.com site with admin rights ( for example a personal site ).
- a wordpress.com site you don't own.
- a self-hosted site you own ( you can use jurassic.ninja ).

1. Make sure you have Jetpack from AppStore installed and logged in.
2. Build and run this PR and log in.
3. Head to **My Site > Menu** screen.
4. **Verify** the rows are grouped in **Content**, **Traffic** and **Maintenance** sections.

## Regression Notes
1. Potential unintended areas of impact
A potential regression bug could be a row visible in older versions and no longer visible in this version or rows that weren't visible before and are now visible. This PR should be tested with wp.com and self-hosted sites you own and other sites you don't own.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and comparing with the version in AppStore.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 1]
```diff
@@ -809,22 +1032,30 @@ - (void)configureTableViewData
...
+        }
+
+        [marr addObject:[self publishTypeSectionViewModel]];
+
+        if ([self shouldAddPersonalizeSection]) {
+            [marr addObject:[self personalizeSectionViewModel]];
+        }
+
+        [marr addObject:[self configurationSectionViewModel]];
+        [marr addObject:[self externalSectionViewModel]];
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m:1052]: The logic under `if ([AppConfiguration isWordPress])` is the existing logic. The menu reorg is only meant for Jetpack.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 2]
```diff
@@ -809,22 +1032,30 @@ - (void)configureTableViewData
...
+    } else {
+        [marr addObject:[self contentSectionViewModel]];
+        [marr addObject:[self trafficSectionViewModel]];
+        [marr addObjectsFromArray:[self maintenanceSectionViewModel]];
     }
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m:1057]: This is the new menu configuration logic for Jetpack.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 3]
```diff
@@ -727,6 +727,229 @@ - (UIView *)tableView:(UITableView *)tableView viewForFooterInSection:(NSInteger
...
+        [weakSelf showViewAdmin];
+        [weakSelf.tableView deselectSelectedRowWithAnimation:YES];
+    }];
+    UIImage *image = [[UIImage gridiconOfType:GridiconTypeExternal withSize:CGSizeMake(BlogDetailGridiconAccessorySize, BlogDetailGridiconAccessorySize)] imageFlippedForRightToLeftLayoutDirection];
+    UIImageView *accessoryView = [[UIImageView alloc] initWithImage:image];
+    accessoryView.tintColor = [WPStyleGuide cellGridiconAccessoryColor]; // Match disclosure icon color.
+    row.accessoryView = accessoryView;
+    row.showsSelectionState = NO;
+    return row;
+}
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m:951]: These "row" properties are extracted from the existing `sectionViewModel` methods. The definition of these rows was embedded in those methods and I needed to re-use the same rows in the new `sectionViewModel` [methods I added.](https://github.com/wordpress-mobile/WordPress-iOS/pull/20476/files#r1157860414)

**Comment by osullivanchris** [comment]
@salimbraksa hey I installed the build and compared the menus across a few sites:
- self hosted
- dotcom
- p2

Almost everything looked as expected. Nice job.

I only found one small discrepency. In the site below, I am not an Admin. In the old version, I could see 'Pages' in the menu. In the new version, I can't see Pages. I am not sure what the correct behaviour should be, but it isn't a deliberate change we're making here. 

Screenshot 2023-04-06 at 10 59 12

**Review by osullivanchris — APPROVED** [review]

**Review by alpavanoglu — APPROVED** [review]

**Comment by mokagio** [comment]
<details>
<summary>I made a wrong assumption that resulted in an unnecessary merge. Leaving it here for reference.</summary>

The PR has been approved but couldn't be merged because it was based on an old `trunk` commit which had a set of CI checks that is no longer in use:

image

I [merged `trunk`](https://github.com/wordpress-mobile/WordPress-iOS/pull/20476/commits/7b2a35b6e5565ee179295e8346816fbbc29902d4) to address that and will enable auto-merge. This was done in the context of getting ahead on the upcoming 22.2 code freeze, as per this PR's milestone.

</details>

This PR couldn't auto-merge because of tests failing _to build_. 4dc21ca fixed the build on my machine and the changed tests still passed. Let's see what CI does.

**Pull request merged**